### DISCUSSION TITLE
Unable to Sign Into ArcGIS Assistant

### DISCUSSION THREAD

**Discussion by Hels7** [EsriPS/arcgis-assistant-feedback#70] [Q&A]
Hi everyone,
I recently experienced a publishing error on a feature layer which our EB app is built on, and want to use ArcGIS Assistant to find out the cause.  
When I sign into ArcGIS assistant it lets me run through the sign in process but circles me back to the main sign-in page each time, i have never been able to get in.  
We use SAML sign in with our work email addresses, which I have tired to use, but I also created a personal ArcGIS log in and tried that without success. I have also check with our IT dept, the URL is not being blocked.
I asked Esri for assistance and they suggested I ask here.  Appreciate any help/ suggestions you can offer.
I'm not a developer, just a project manager who is very new to GIS - but also loving it.
Thanks Kindly, Hels

**Comment by rlesperance** [comment]
From what I understand there's an issue with the OAuth.  Can you clarify if you are working in Enterprise or ArcGIS Online?  If it's Online, there's a setting under the Apps section in security where you can disable people being able to authenticate through another app unless it's explicitly registered.  You can add the app and register it, but at that point Assistant would need an App ID.  That workflow is available for connecting to Enterprise, but not ArcGIS Online.    I think an improvement should be made there, but for now you'll have to turn that protection off while using the tool.

**Comment by Hels7** [comment]
Hi, thanks for getting back to me.  We are using ArcGIS online..  I am trying to understand the implications of disabling the protection you mention but haven’t got very far yet. What kind of apps do you mean?

**Comment by nnnnoodle** [comment]
@Hels7 It sounds like I am having a similar issue trying to connecting ArcGIS Online Assistant with Enterprise. I have connected with our AGOL OK but when I try the connect to Enterprise the login process loops back round to the ArcGIS Online Assistant homepage without adding Enterprise as a connection.

Might someone be able to confirm that I have configured the Application with Enterprise correctly?

![image](https://github.com/user-attachments/assets/88a42c07-c073-4b48-a745-500a53e5199b)

![image](https://github.com/user-attachments/assets/03bc395f-a064-4ead-b10d-450e9931c6a5)

If this is all OK what else must I do to get this work?

Many thanks,

Nick

**Reply by tshewring-he** [reply]
I am getting exactly the same issue using Enterprise 11.3. Have you been able to connect to Enterprise? Thanks

**Reply by nnnnoodle** [reply]
Thank you @tshewring-he for the reminder to update this thread! Yes, problem solved...

In Portal go to Organisation > Settings > Security > Allow origins and use the "Add domain" button to add https://assistant.esri-ps.com as a new CORS domain.

Please take care to not include the closing forward slash. If you add the domain using **https://assistant.esri-ps.com/** the new CORS config won't work, it has to be **https://assistant.esri-ps.com** without that last forward slash, as shown below:

![image](https://github.com/user-attachments/assets/bc8657ae-0be9-4569-bf67-f4ea0423fa76)

**Reply by tshewring-he** [reply]
Hi @nnnnoodle thanks for the update. Unfortunately this doesn't resolve the issue for me. I have added https://assistant.esri-ps.com to Security>Allow origins but when I sign in to the ArcGIS Assistant (using Sign in with ArcGIS Enterprise) - I get the Request for Permission screen -> I select 'Allow' and I am still caught in the login process loop -  going back round to the 'ArcGIS Assistant Beta | 20230927.2 Sign in' homepage without adding Enterprise as a connection.

@nnnnoodle - Update - I've just got it working but I have to use the 7443 address for Enterprise Portal (bypass WebAdaptor) when signing in via the ArcGIS Assistant. This doesn't require an entry in Security>Allow origins

**Reply by nicsi76** [reply]
Thanks @nnnnoodle for the hint with removing the ending slash from the link - that finally made it work.

**Comment by nnnnoodle** [comment]
...side note, all the documentation talks about the use of an App ID but at Enterprise 11.3 this seems to have been renamed as a Client ID. Might it be possible for the documentation to be updated to use the new nomenclature? Thanks in advance!

**Comment by DannyGant** [comment]
Good afternoon all.  Getting the "Your active account is not authenticated." error when trying to auth into Assistant.  Can sign in just fine to the Org that my credentials belong to.  Can also sign into Assistant using credentials from another Org just fine.  ESRI doesn't support, looking for help here.  I would note the suggestion of adding the url to "Allow Origins" didn't rectify my issue.

Dev Tools:
![image](https://github.com/user-attachments/assets/c013cf31-925c-4202-b106-c4e81ca69d49)
